pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -176,6 +181,16 @@ func (pt PipelineTask) Deps() []string {
 			}
 		}
 	}
+	// Add any dependents from when expressions
+	for _, whenExpression := range pt.WhenExpressions {
+		expressions, ok := GetVarSubstitutionExpressionsForWhenExpression(whenExpression)
+		if ok {
+			resultRefs := NewResultRefs(expressions)
+			for _, resultRef := range resultRefs {
+				deps = append(deps, resultRef.PipelineTask)
```

Comment:
What happens for a task with a param and `when` expression referring to the same task result of the parent task `deps` ends up having duplicate entry for the same parent task?

Reply:
it'd add duplicate entry here, the logic right above for adding dependency from `runafter`, `resources`, `conditions` and `results` doesn't check for uniqueness of the dependency...maybe that's handled later, maybe in the `dag`...will look into it tomorrow